package/libpiksi/libpiksi/src/settings.c

```diff
@@ -1228,33 +204,19 @@ settings_ctx_t *settings_create(void)
     return ctx;
   }
 
-  /* Register standard types */
-  settings_type_t type;
+  ctx->setreg = setreg_create();
 
-  if (type_register(ctx, int_to_string, int_from_string, NULL, NULL, &type) != 0) {
-    destroy(&ctx);
-    return ctx;
-  }
-  assert(type == SETTINGS_TYPE_INT);
-
-  if (type_register(ctx, float_to_string, float_from_string, NULL, NULL, &type) != 0) {
-    destroy(&ctx);
-    return ctx;
-  }
-  assert(type == SETTINGS_TYPE_FLOAT);
-
-  if (type_register(ctx, str_to_string, str_from_string, NULL, NULL, &type) != 0) {
-    destroy(&ctx);
-    return ctx;
-  }
-  assert(type == SETTINGS_TYPE_STRING);
+  setreg_api_t api = {0};
+  api.ctx = (void *)ctx;
```

Comment:
Here I think I would much rather create the `api_t` object and pass that into the `create` method for `libsettings` and retain the naming of the `settings_ctx_t` and other api names, such that `libpiksi/settings.c` need not wrap or simply forward on redundant calls.

Reply:
I am not 100% if we are talking about same thing, but in this buildroot context the API wrappers are indeed quite naive, but in the FW wrappers there's actually something done besides forwarding the call: https://github.com/swift-nav/piksi_firmware_private/pull/2452/files#diff-c51ec15c7690cd5a6e83019fd69a14eeR64